plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePartitionsTable.java

```diff
@@ -0,0 +1,455 @@
+    {
+        public static class Builder
+        {
+            private final TypeManager typeManager;
+            private long fileCount;
+            private long size;
+            private final Map<String, ColumnStatistics> columnStatistics = new HashMap<>();
+            private final Map<String, Optional<Long>> nullCounts = new HashMap<>();
+            private final Map<String, Optional<Long>> nanCounts = new HashMap<>();
+
+            public Builder(TypeManager typeManager)
+            {
+                this.typeManager = requireNonNull(typeManager, "typeManager is null");
+            }
+
+            public void acceptAddFileEntry(AddFileEntry addFileEntry)
+            {
+                fileCount++;
+                size += addFileEntry.getSize();
+            }
+
+            public DeltaLakePartitionStatistics build()
+            {
+                ImmutableMap.Builder<String, Object> minValues = ImmutableMap.builder();
+                ImmutableMap.Builder<String, Object> maxValues = ImmutableMap.builder();
+
+                columnStatistics.forEach((key, statistics) -> {
+                    statistics.getMin().ifPresent(min -> minValues.put(key, min));
+                    statistics.getMax().ifPresent(max -> maxValues.put(key, max));
+                });
+
+                Map<String, Long> nullCounts = this.nullCounts.entrySet().stream()
+                        .filter(entry -> entry.getValue().isPresent())
+                        .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().orElseThrow()));
+
+                Map<String, Long> nanCounts = this.nanCounts.entrySet().stream()
+                        .filter(entry -> entry.getValue().isPresent())
+                        .collect(toImmutableMap(Map.Entry::getKey, entry -> entry.getValue().orElseThrow()));
+
+                return new DeltaLakePartitionStatistics(fileCount, size, minValues.buildOrThrow(), maxValues.buildOrThrow(), nullCounts, nanCounts);
+            }
+
+            private void updateNullCountStats(String key, Optional<Long> nullCount)
+            {
+                // If one file is missing nullCounts for a column, invalidate the estimate
+                nullCounts.merge(key, nullCount, (existingCount, newCount) ->
+                        existingCount.isPresent() && newCount.isPresent() ? Optional.of(existingCount.get() + newCount.get()) : Optional.empty());
+            }
+
+            private void updateNanCountStats(String key, Optional<Long> nanCount)
+            {
+                // If one file is missing nanCounts for a column, invalidate the estimate
+                nanCounts.merge(key, nanCount, (existingCount, newCount) ->
+                        (existingCount.isPresent() && newCount.isPresent()) ? Optional.of(existingCount.get() + newCount.get()) : Optional.empty());
+            }
+
+            private void updateMinMaxStats(
```

Comment:
Unused methods.

Reply:
Implementation for `data` column is still in progress. These will be used.

Reply:
> These will be used.

in this PR or follow-up PR?

Reply:
This PR.